noxfile.py

```diff
@@ -134,6 +134,15 @@ def format(session):
     )
 
 
+@nox.session(python=DEFAULT_PYTHON_VERSION)
+def mypy(session):
+    """Verify type hints are mypy compatible."""
+    session.install("-e", ".")
+    session.install("mypy", "types-setuptools")
+    # TODO: also verify types on tests and logging_v2, big undertakings
```

Comment:
are those big undertakings? logging_v2 should already have valid types in place. And I'd say tests are optional